lib/sqlalchemy/sql/elements.py

```diff
@@ -1837,7 +1853,7 @@ def _dedupe_anon_label_idx(self, idx: int) -> str:
             return self._dedupe_anon_tq_label_idx(idx)
 
     @property
-    def _proxy_key(self):
+    def _proxy_key(self) -> Optional[str]:
```

Comment:
Changing `@property` to `@HasMemoized.memoized_attribute` (superclass definition has it) removes the error. This isn't the only case, guessing there's an access-related conflict somewhere between `@property`, `@HasMemoized.memoized_attribute` and `@util.memoized_property` from mypy's point of view.

Reply:
the `@property` situation with mypy is very difficult.    it's not in general safe to change a `@property` to a memoized so I'd have to look at this.